src/get-configvals.c

```diff
@@ -0,0 +1,44 @@
+/* 
+* Copyright (c) 2017, SingularityWare, LLC., Inc. All rights reserved.
+* 
+* This software is licensed under a 3-clause BSD license.  Please
+* consult LICENSE file distributed with the sources of this project regarding
+* your rights to use or distribute this software.
+* 
+*/
+
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <string.h>
+
+#include "util/util.h"
+#include "util/config_parser.h"
+
+#ifndef SYSCONFDIR
+#error SYSCONFDIR not defined
+#endif
+
+int main(int argc, char **argv) {
+
+    if ( argc < 2 ) {
+        printf("USAGE: %s [key]\n", argv[0]);
+        exit(0);
+    }
+
+    char *key = strdup(argv[1]);
+
+    singularity_config_init();
+
+    /* 
+    If the key does not exist in the singularity.conf file, this is just 
+    going to return null char.  The function was originally designed to return
+    the default value if the value does not exist in the conf file, but I don't 
+    know how to do that using only strings, and the key needs to be based on 
+    user input, not hardcoded. 
+    */
+    printf("%s\n", _singularity_config_get_value_impl(key, "\0"));
```

Comment:
Just use `singularity_config_get_value()` here.

Reply:
Unless I'm mistaken, I can't use `singularity_config_get_value()`.  It only works with hard-coded variables right?  I cant just pass some arbitrary string out of bash to it.

Reply:
Isn't there some avenue to get any variable that starts with `SINGULARITY_` ? Not sure how it works in the C but I know they magically show up in the bash, haha.

Reply:
I think this one pulls from env --> https://github.com/singularityware/singularity/blob/adf5259ac94d7b213c649f073dcda32459b0c8b7/src/util/registry.c#L103 and the previous is from the config.

Reply:
oh and I think the comment above is saying to use the higher level function to that one (without the _)

Reply:
It's not a higher level function.  It's a macro that is designed to only work with variables that are defined at build time.  Unless I'm mistaken.  That means I can't just pass an arbitrary character.  That is why I'm using `_singularity_config_get_value_impl()`.  If there is a way to use  `singularity_registry_get()` please show me.

Reply:
I don't know C well enough to be able to give a confident answer. I can say that a long time ago I started (almost the same named script) but did try using the registry function: https://github.com/vsoch/singularity/blob/961fcf50a264c8574071adb311bd78daf2674e13/src/get-config.c I needed some mentoring with regard to what to try next so it never went anywhere from there. Anyway, I'm sure @gmkurtzer will be able to give some insight here!

Reply:
That's hilarious!  It's almost identical.  If I had to wager a guess I would say that get-config.c doesn't compile and it complains that key is not defined at compile time (or some such error in c-speak).  Am I right?

Reply:
Haha, no idea, I can't remember... this was back in July!